pandas/tests/frame/methods/test_reset_index.py

```diff
@@ -183,6 +183,28 @@ def test_reset_index_name(self):
         assert return_value is None
         assert df.index.name is None
 
+    def test_reset_index_rename(self, float_frame):
+        # GH 6878
+        rdf = float_frame.reset_index(names="new_name")
+        exp = Series(float_frame.index.values, name="new_name")
+        tm.assert_series_equal(rdf["new_name"], exp)
+
+    def test_reset_index_rename_multiindex(self, float_frame):
+        # GH 6878
```

Comment:
are there times when this *should* raise? e.g. not a list-like to names or have a length mismatch i would think.